holidays/countries/ethiopia.py

```diff
@@ -48,72 +54,61 @@ def _populate(self, year):
+        self._add_christmas_day("ገና/Ethiopian X-Mas")
 
         # Ethiopian Ephiphany
-        self[date(year, JAN, 19)] = "ጥምቀት/Ephiphany"
+        self._add_epiphany_day("ጥምቀት/Ephiphany")
 
         # Ethiopian Good Friday
-        easter_date = easter(year, EASTER_ORTHODOX)
-        self[easter_date + rd(days=-2)] = "ስቅለት/Ethiopian Good Friday"
+        self._add_good_friday("ስቅለት/Ethiopian Good Friday")
 
         # Ethiopian  Easter - Orthodox Easter
-        self[easter_date] = "ፋሲካ/Ethiopian Easter"
+        self._add_easter_sunday("ፋሲካ/Ethiopian Easter")
 
         # Adwa Victory Day
         if year > 1896:
-            self[date(year, MAR, 2)] = "አድዋ/Victory of Adwa"
+            self._add_holiday("አድዋ/Victory of Adwa", MAR, 2)
 
         # Labour Day
-        self[date(year, MAY, 1)] = "የሰራተኞች ቀን/Labor Day"
+        self._add_labour_day("የሰራተኞች ቀን/Labor Day")
 
         # Patriots Day
         if year > 1941:
-            self[date(year, MAY, 5)] = "የአርበኞች ቀን/Patriots Day"
+            self._add_holiday("የአርበኞች ቀን/Patriots Day", MAY, 5)
 
         # Derg Downfall Day
         if year > 1991:
-            self[
-                date(year, MAY, 28)
-            ] = "ደርግ የወደቀበት ቀን/Downfall of Dergue regime"
+            self._add_holiday(
+                "ደርግ የወደቀበት ቀን/Downfall of Dergue regime", MAY, 28
+            )
 
         # Downfall of King. Hailesilassie
         if year < 1991 and year > 1974:
             if self.ethiopian_isleap(year):
-                self[date(year, SEP, 13)] = "ደርግ የመጣበት ቀን/Formation of Dergue"
+                self._add_holiday("ደርግ የመጣበት ቀን/Formation of Dergue", SEP, 13)
             else:
-                self[date(year, SEP, 12)] = "ደርግ የመጣበት ቀን/Formation of Dergue"
+                self._add_holiday("ደርግ የመጣበት ቀን/Formation of Dergue", SEP, 12)
 
         # Eid al-Fitr - Feast Festive
-        # date of observance is announced yearly, This is an estimate since
-        # having the Holiday on Weekend does change the number of days,
-        # deceided to leave it since marking a Weekend as a holiday
-        # wouldn't do much harm.
-        for date_obs in _islamic_to_gre(year, 10, 1):
-            hol_date = date_obs
-            self[hol_date] = "ኢድ አልፈጥር/Eid-Al-Fitr"
+        self._add_eid_al_fitr_day("ኢድ አልፈጥር/Eid-Al-Fitr")
 
         # Eid al-Adha - Scarfice Festive
         # date of observance is announced yearly
-        for date_obs in _islamic_to_gre(year, 12, 9):
-            hol_date = date_obs
-            self[hol_date + rd(days=1)] = "አረፋ/Eid-Al-Adha"
-
-        # Prophet Muhammad's Birthday - (hijari_year, 3, 12)
-        for date_obs in _islamic_to_gre(year, 3, 12):
-            hol_date = date_obs
-            self[hol_date + rd(days=1)] = "መውሊድ/Prophet Muhammad's Birthday"
+        self._add_eid_al_adha_day("አረፋ/Eid-Al-Adha")
+
+        # Prophet Muhammad's Birthday.
+        self._add_mawlid_day_two("መውሊድ/Prophet Muhammad's Birthday")
```

Comment:
```suggestion
        self._add_mawlid_day("መውሊድ/Prophet Muhammad's Birthday")
```

Reply:
Doesn't seem to be an error here. Technically it's a Mawlid day 2 date in Ethiopia (tests agree). I'll look into fixing/clarifying this in a different PR if needed.

Reply:
I didn't see `+ rd(days=1)` there, sorry.